DQM/HGCal/plugins/HGCalTriggerClient.cc

```diff
@@ -84,103 +204,434 @@ void HGCalTriggerClient::analyze(const edm::Event& iEvent, const edm::EventSetup
+      if (verbose_ >= 2)
+        out << "Event : " << iEvent.id().event() << ", Event size do not match between trigger RO header "
+            << daq_event_size[i] << " and " << i << "-th 0xfecafe word " << size_in_cafe[i] << "\n";
+      h_ECONTRawDataErrors->Fill(2);
+      return;
     }
-    out << std::dec << std::setfill(' ');
-  }
-
-  // ====== TODO: ECON-T unpacker ======
-  // ...
-  // ====== TODO: ECON-T unpacker ======
-
-  // //read flagged ECON-D list
-  // const auto& flagged_econds = iEvent.getHandle(econdQualityToken_);
-  // if (flagged_econds.isValid()) {
-  //   for (auto econd : *flagged_econds) {
-  //     HGCalElectronicsId eleid(econd.eleid);
-  //     MonitorKey_t logiKey(eleid.zSide(), eleid.fedId(), eleid.captureBlock(), eleid.econdIdx());
-  //     MonitorKey_t k = module_keys_[logiKey];
-  //     int ibin = modbins_[k];
-  //     p_econdpayload->Fill(ibin, econd.payload);
-  //     p_econdcbquality->Fill(ibin, econd.cbflags);
-  //     if (econd.cbFlag())
-  //       p_econdquality->Fill(ibin, 0);
-  //     if (econd.htFlag())
-  //       p_econdquality->Fill(ibin, 1);
-  //     if (econd.eboFlag())
-  //       p_econdquality->Fill(ibin, 2);
-  //     if (econd.matchFlag())
-  //       p_econdquality->Fill(ibin, 3);
-  //     if (econd.truncatedFlag())
-  //       p_econdquality->Fill(ibin, 4);
-  //     if (econd.wrongHeaderMarker())
-  //       p_econdquality->Fill(ibin, 5);
-  //     if (econd.payloadOverflows())
-  //       p_econdquality->Fill(ibin, 6);
-  //     if (econd.payloadMismatches())
-  //       p_econdquality->Fill(ibin, 7);
-  //   }
-  // }
-}
+  }
 
-void HGCalTriggerClient::bookHistograms(DQMStore::IBooker& ibook, edm::Run const& run, edm::EventSetup const& iSetup) {
-  // //create module keys
-  // auto moduleInfo = iSetup.getData(moduleInfoToken_);
-  // module_keys_ = moduleInfo.getAsSimplifiedModuleLocatorMap(true);
-  // size_t nmods = module_keys_.size();
-  // LogDebug("HGCalTriggerClient") << "Read module info with " << nmods << " entries";
+  if (daq_nbx[0] != daq_nbx[1]) {
+    if (verbose_ >= 1)
+      out << "Event : " << iEvent.id().event() << ", Bx size do not match between packed " << daq_nbx[0]
+          << " and unpacked data " << daq_nbx[1] << std::endl;
+    //nofNbxMisMatches++;
+    h_ECONTRawDataErrors->Fill(3);
+    return;
+  }
+
+  //////////////////// NOTE : The following raw decoding is only valid for STC mode ///////////////////////
+
+  int bx_index = -1.0 * int(daq_nbx[0]);
+  const int maxnbx = (2 * daq_nbx[0] + 1);  //In case of raw input
+  uint32_t energy_raw[2][maxnbx][12];       //2 : LSB/MSB, 12 : STC
+  uint32_t loc_raw[2][maxnbx][12];          //2 : LSB/MSB, 12 : STC
+  uint32_t bx_raw[2][maxnbx][12];           //2 : LSB/MSB, 15 : STC
+  uint32_t packet[4];                       //a bx packet constitutes 4 64 bit words
+  uint32_t packet_locations[12];
+  uint64_t packet_energies[12];
+
+  for (int iect = 0; iect < 2; iect++) {
```

Comment:
This is not needed if you [use `{0}`](https://stackoverflow.com/questions/31114644/initialize-multidimensional-array-with-zeros).

Reply:
Necessary modifications have been applied (see [8165d07](https://github.com/CMS-HGCAL/cmssw/pull/86/commits/8165d077af54e5ab2fcceb6c538b92669bbe113d))